bin/basket.js

```diff
@@ -35,9 +35,36 @@ basketQueue.on(
           }
         },
         function (err, res, body) {
+          // Log and retry for network-level errors.
           if (err) {
-            return log.error({ op: 'basketRequest', err: err })
+            log.error(
+              {
+                op: 'basketRequest',
+                uid: message.uid,
+                locale: message.locale,
+                err: err
+              }
+            )
+            return
           }
+          // Log all HTTP-level errors.
+          if (res.statusCode < 200 || res.statusCode >= 300) {
+            log.error(
+              {
+                op: 'basketRequest',
+                uid: message.uid,
+                locale: message.locale,
+                status: res.statusCode,
+                body: body
+              }
+            )
+            // Retry for 500-level errors, drop for 400-level.
+            if (res.statusCode < 500) {
```

Comment:
So, current queue configuration is the default 'Default Visibility Timeout: 5 minutes', and 'Message Retention Period: 	14 days'. I know we discussed this, but don't recall the final opinion (if there was one). At any rate, I think a longer visibility timeout would be better and lessens the chance of getting into a pathological retry state. Could also change the visibility dynamically with ChangeMessageVisibility